swirl_lm/example/fire/fire_uq.py

```diff
@@ -0,0 +1,167 @@
+  'random_seed_uq',
+  0,
+  'The random seed used for sampling the uncertain parameters.'
+)
+_N_SAMPLES_UQ = flags.DEFINE_integer(
+  'n_samples_uq',
+  100,
+  'The number of samples collected for uncertainty quantification.'
+)
+_FUEL_DENSITY_LB = flags.DEFINE_float(
+  'fuel_density_lb',
+  0.2,
+  'Lower bound of fuel density for uncertainty quantification.'
+)
+_FUEL_DENSITY_UB = flags.DEFINE_float(
+  'fuel_density_ub',
+  3.0,
+  'Upper bound of fuel density for uncertainty quantification.'
+)
+_MOISTURE_CONTENT_LB = flags.DEFINE_float(
+  'moisture_content_lb',
+  0.03,
+  'Lower bound of moisture content for uncertainty quantification.'
+)
+_MOISTURE_CONTENT_UB = flags.DEFINE_float(
+  'moisture_content_ub',
+  0.12,
+  'Upper bound of moisture content for uncertainty quantification.'
+)
+_WIND_SPEED_LB = flags.DEFINE_float(
+  'wind_speed_lb',
+  5.0,
+  'Lower bound of wind speed for uncertainty quantification.'
+)
+_WIND_SPEED_UB = flags.DEFINE_float(
+  'wind_speed_ub',
+  12.0,
+  'Upper bound of wind speed for uncertainty quantification.'
+)
+_MODIFY_INDIVIDUAL = flags.DEFINE_bool(
+  'modify_indiviual',
+  False,
+  'Whether to modify the different values used in uncertainty quantification'
+  'one at a time. This produces only 4 trajectories and should only be used'
+  'for validation.'
+)
+_READ_UQ_FILE = flags.DEFINE_bool(
+  'read_uq_file',
+  False,
+  'Whether to read the UQ-parameters from a file',
+)
+_UQ_FILENAME = flags.DEFINE_string(
+  'uq_filename',
+  './large_scale_uq_params.npy',
+  'The location of the .npy file that contains the uq values'
+)
+_UQ_START_ID = flags.DEFINE_integer(
+  'uq_start_id',
+  0,
+  'The start id for uq.'
+)
+_UQ_END_ID = flags.DEFINE_integer(
+  'uq_end_id',
+  19,
+  'The end id for uq.'
+)
+
+
+class FireUQSampler:
+  def __init__(self):
+    self.sampler = np.random.default_rng(_RANDOM_SEED_UQ.value)
+    self.start_id = _UQ_START_ID.value
+    self.end_id = _UQ_END_ID.value
+    if _MODIFY_INDIVIDUAL.value:
+      logging.warn(
+        'Modifying uncertain parameters one at a time. Only generating 4'
+        'trajectories (1 trajectory for baseline and 1 for each varied'
+        'parameter). Ignores n_samples_uq flag.'
+      )
+    pass
+  
+  def _get_norm_samples(self, n_samples, n_uq_vars=3):
+    norm_samples = self.sampler.random(size=(n_samples, n_uq_vars), dtype=np.float32)
+    return norm_samples
+
+  def _shift_scale(self, lower_bound, upper_bound, norm_samples):
+    """
+    Shift and scales norm_samples to the interval [lower_bound, uppber_bound].
+    We do not use `self.sampler.uniform` for compatibility with previous
+    version of random number generation and for keeping the first n uq-samples
+    the same.
+    """
+    return (upper_bound - lower_bound) * norm_samples + lower_bound
+
+  def number_of_samples(self):
+    if _MODIFY_INDIVIDUAL.value:
+      return 4
+    elif _READ_UQ_FILE.value:
+      uq_values = np.load(_UQ_FILENAME.value)
```

Comment:
This needs to go through gfile because np.load doesn't necessarily work on other file systems (e.g., cloud buckets). Here's an example of doing that: https://github.com/google-research/swirl-lm/blob/809b8fbd3962d9eb1c6e55871eaf4ba5eac72687/swirl_lm/example/fire/terrain_utils.py#L75

I think there's one more place where you call np.load on a filename.

Reply:
Thanks; I was not aware of this. Changed to use gfile in [6411a90](https://github.com/google-research/swirl-lm/pull/3/commits/6411a9091deab73d0cade6e8445fab4ac58a9139)